hbase-external-blockcache/src/main/java/org/apache/hadoop/hbase/io/hfile/MemcachedBlockCache.java

```diff
@@ -139,17 +165,24 @@ public Cacheable getBlock(BlockCacheKey cacheKey, boolean caching, boolean repea
     try (Scope traceScope = span.makeCurrent()) {
       result = client.get(cacheKey.toString(), tc);
     } catch (Exception e) {
-      // Catch a pretty broad set of exceptions to limit any changes in the memecache client
+      // Catch a pretty broad set of exceptions to limit any changes in the memcache client
       // and how it handles failures from leaking into the read path.
-      if (LOG.isDebugEnabled()) {
-        LOG.debug(
-          "Exception pulling from memcached [ " + cacheKey.toString() + " ]. Treating as a miss.",
-          e);
+      if (
+        (e instanceof OperationTimeoutException) || ((e instanceof RuntimeException)
+          && (e.getCause() instanceof OperationTimeoutException))
+      ) {
+        timeoutCount.incrementAndGet();
+        if (LOG.isDebugEnabled()) {
+          LOG.debug("Timeout getting key " + cacheKey.toString(), e);
+        }
+      } else {
+        if (LOG.isDebugEnabled()) {
+          LOG.debug("Exception getting key " + cacheKey.toString(), e);
+        }
       }
```

Comment:
We need to increment `cacheErrorCount` correct?

Reply:
Fixed, thanks for pointing this out.